lxd/certificates.go

```diff
@@ -57,7 +58,27 @@ func (d *Daemon) verifyAdminPwd(password string) bool {
 	return true
 }
 
+type certificateResponse struct {
+	Certificate string `json:"certificate"`
+	Fingerprint string `json:"fingerprint"`
+	Type        string `json:"type"`
+}
+
 func certificatesGet(d *Daemon, r *http.Request) Response {
+	recursion := d.isRecursionRequest(r)
+
+	if recursion {
+		certResponses := []certificateResponse{}
+		for _, cert := range d.clientCerts {
+			resp := certificateResponse{}
+			resp.Fingerprint = shared.GenerateFingerprint(&cert)
+			resp.Certificate = base64.StdEncoding.EncodeToString(cert.Raw)
+			resp.Type = "client"
+			certResponses = append(certResponses, resp)
```

Comment:
As I said below, it'd probably be best to do like the other recursion code paths and rather than duplicate the code for the entry, call a shared function which returns the entry.